virttest/test_setup.py

```diff
@@ -2029,16 +2030,55 @@ def stop(self):
         self._compress_log()
 
 
-def disable_smt():
+def disable_smt(params=None, cleanup=False):
     """
     Checks whether smt is on, if so disables it in PowerPC system
-    This function is used in env_process to check & disable smt in Power8
+    This function is used in env_process and in libvirt.py to check
+    & disable smt in Power8 for local and remote machine respectively.
+
+    :param params: Test params dict for remote machine details to turn off smt
+    :param cleanup: If True, revert back to default setting for cleanup
     """
-    smt_output = process.system_output("ppc64_cpu --smt", shell=True).strip()
-    if smt_output != "SMT is off":
+    cmd = "ppc64_cpu --smt"
+    smt_output = ""
+    if params:
```

Comment:
better not to use params as the parameter to check for remote, have a separate flag for remote host